designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/RemoteEnvironment.scala

```diff
@@ -49,10 +56,9 @@ trait RemoteEnvironment {
   def processVersions(processName: ProcessName)(implicit ec: ExecutionContext): Future[List[ScenarioVersion]]
 
   def migrate(
-      localProcess: ScenarioGraph,
-      processName: ProcessName,
-      parameters: ScenarioParameters,
-      isFragment: Boolean
+      processingMode: ProcessingMode,
+      engineSetupName: EngineSetupName,
+      scenarioToMigrate: ScenarioWithDetailsForMigrations
```

Comment:
1. Let's remove this temporary class and extract minimal set of information that we need for migration
2. Let's consider various combinations of backward/forward incompatible changes and versioning schemas